src/Shared/Targets/Wrappers/AspNetBufferingTargetWrapper.cs

```diff
@@ -0,0 +1,305 @@
+        public AspNetBufferingTargetWrapper(Target wrappedTarget, int bufferSize)
+        {
+        
+            WrappedTarget      = wrappedTarget;
+            BufferSize         = bufferSize;
+            GrowBufferAsNeeded = true;
+        }
+
+        /// <summary>
+        /// Gets or sets the number of log events to be buffered.
+        /// </summary>
+        /// <docgen category='Buffering Options' order='100' />
+        [DefaultValue(100)]
+        public int BufferSize { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether buffer should grow as needed.
+        /// </summary>
+        /// <value>A value of <c>true</c> if buffer should grow as needed; otherwise, <c>false</c>.</value>
+        /// <remarks>
+        /// Value of <c>true</c> causes the buffer to expand until <see cref="BufferGrowLimit"/> is hit,
+        /// <c>false</c> causes the buffer to never expand and lose the earliest entries in case of overflow.
+        /// </remarks>
+        /// <docgen category='Buffering Options' order='100' />
+        [DefaultValue(false)]
+        public bool GrowBufferAsNeeded { get; set; }
+
+        /// <summary>
+        /// Limits the amount of slots that the buffer should grow
+        /// </summary>
+        private int _bufferGrowLimit;
+
+        /// <summary>
+        /// Gets or sets the maximum number of log events that the buffer can keep.
+        /// </summary>
+        /// <docgen category='Buffering Options' order='100' />
+        public int BufferGrowLimit
+        {
+            get => _bufferGrowLimit;
+
+            set
+            {
+                _bufferGrowLimit = value;
+                GrowBufferAsNeeded = (value >= BufferSize);
+            }
+        }
+
+        /// <summary>
+        /// Context for DI
+        /// </summary>
+        private IHttpContextAccessor _httpContextAccessor;
+
+        /// <summary>
+        /// Provides access to the current request HttpContext.
+        /// </summary>
+        /// <returns>HttpContextAccessor or <c>null</c></returns>
+        internal IHttpContextAccessor HttpContextAccessor
+        {
+            get => _httpContextAccessor ?? (_httpContextAccessor = RetrieveHttpContextAccessor());
+            set => _httpContextAccessor = value;
+        }
+
+        private IHttpContextAccessor RetrieveHttpContextAccessor()
+        {
+#if ASP_NET_CORE
+            return ServiceLocator.ResolveService<IHttpContextAccessor>(ResolveService<IServiceProvider>(), LoggingConfiguration);
+#else
+            return new DefaultHttpContextAccessor();
+#endif
+        }
+
+        /// <summary>
+        /// Must be called by the HttpModule/Middleware upon starting.
+        /// This creates the dictionary in the HttpContext.Items with the proper key.
+        /// </summary>
+        /// <param name="context"></param>
+        internal static void Initialize(HttpContextBase context)
```

Comment:
Could this be renamed to `BeginRequest` ?

Reply:
Completed